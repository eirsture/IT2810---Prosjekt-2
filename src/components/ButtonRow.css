
.buttons {
    margin: 0;
    padding: 0;
    display:flex;
    flex-direction: row;
    flex:1;
    height: 100%;
    width: 100%;
    grid-template-rows: 1;
    grid-template-columns: repeat(4,1fr);
    background-color: gray
}