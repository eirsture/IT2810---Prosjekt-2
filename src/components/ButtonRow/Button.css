
.title {
    font-size: 20px;
}

.button {
    margin: 0 5px;
    height: 75%;
    width: 100%;    
    cursor: pointer;
    border: 2px solid rgba(0, 0, 0, 0.25);
    border-bottom: none;
    border-radius: 20px 20px 0 0;
}

.button:hover {
    background-color: coral;
    border-radius: 20px 20px 0 0;
}

.selected {
    background-color: coral;
    border-radius: 20px 20px 0 0;
    box-shadow: 0 1px 6px 0 rgba(0, 0, 0, 0.5);
}

@media only screen and (max-width: 768px){
    .selected {
        box-shadow: none;
        border-bottom: none;
    }
}