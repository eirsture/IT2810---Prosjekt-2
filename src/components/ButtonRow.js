import React from 'react';
import "./ButtonRow.css";
import "./Button.css"

class ButtonRow extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            selectedButton: 1,
        }
    }


    handleButtonPress = (event) => {
        this.setState({selectedButton: parseInt(event.currentTarget.id)})
        this.props.selectedTab(parseInt(event.currentTarget.id))
        this.saveToLocalStorage(event.currentTarget.id)
    }

    saveToLocalStorage = (id) => {
        localStorage.setItem('selectedTab', id)
    }

    componentDidMount = () => {
        const locallyStoredId = localStorage.getItem('selectedTab')
        if (locallyStoredId) {
            this.setState({ selectedButton: locallyStoredId})
        }
        console.log(locallyStoredId)
    }

    render() {
        console.log(this.state.selectedButton)
    return (
        <div className="buttons">{
            [1,2,3,4].map(key => 
            <div 
                key= {key}
                id = {key}
                className={key === this.state.selectedButton ? "selected button" : "hidden button"}
                onClick={this.handleButtonPress}
            >
                <p className="title">
                    {key}
                </p>
            </div>
        )
        }
        </div>
    );
    }
}

export default ButtonRow;