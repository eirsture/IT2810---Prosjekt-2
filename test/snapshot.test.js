import React from 'react'
import renderer from 'react-test-renderer';
import ButtonRow from '../src/components/ButtonRow/ButtonRow.js';
import SidebarContent from '../src/components/Sidebar/sidebarContent.js';
import SidebarSection from '../src/components/Sidebar/sidebarSection.js';

describe("Button Row component", ()=> {
it('renders correctly', () => {
    const tree = renderer
      .create(<ButtonRow/>)
      .toJSON();
    console.log(tree)
    expect(tree).toMatchSnapshot();
  });
it('is correct type', ()=> {
    const tree = renderer
    .create(<ButtonRow/>)
    .toJSON();
    expect(tree.type).toBe('div'); 
    });
});

describe("Sidebar section", ()=> {
    it('renders correctly', () => {
        const tree = renderer
          .create(<SidebarSection categories={["Cat", "Dog", "Horse"]}/>)
          .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('is correct type', ()=> {
        const tree = renderer
        .create(<SidebarSection categories={["Cat", "Dog", "Horse"]}/>)
        .toJSON();
        expect(tree.type).toBe('div'); 
    });
});

describe("Sidebar content", ()=> {
    it('renders correctly', () => {
        const tree = renderer
            .create(<SidebarContent categories={["Cat", "Dog", "Horse"]}/>)
            .toJSON();
        console.log(tree)
        expect(tree).toMatchSnapshot();
        });
    it('contains sidebar content', ()=> {
        const tree = renderer
        .create(<SidebarContent categories={["Cat", "Dog", "Horse"]}/>)
        .toJSON();
        console.log(tree.children)
        expect(tree.children).toHaveLength(3)
    });
});