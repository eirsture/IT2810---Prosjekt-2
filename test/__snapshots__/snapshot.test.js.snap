// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Row component renders correctly 1`] = `
<div
  className="buttons"
>
  <div
    className="selected button"
    id={1}
    onClick={[Function]}
  >
    <p
      className="title"
    >
      1
    </p>
  </div>
  <div
    className="hidden button"
    id={2}
    onClick={[Function]}
  >
    <p
      className="title"
    >
      2
    </p>
  </div>
  <div
    className="hidden button"
    id={3}
    onClick={[Function]}
  >
    <p
      className="title"
    >
      3
    </p>
  </div>
  <div
    className="hidden button"
    id={4}
    onClick={[Function]}
  >
    <p
      className="title"
    >
      4
    </p>
  </div>
</div>
`;

exports[`Sidebar content renders correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="Image"
  >
    <div
      className="sectionContainer"
    >
      <h3>
        Image
      </h3>
      <form>
        <div
          className="radio active"
        >
          <label>
            <input
              checked={true}
              onChange={[Function]}
              type="radio"
              value="Cat"
            />
            Cat
          </label>
        </div>
        <div
          className="radio"
        >
          <label>
            <input
              checked={false}
              onChange={[Function]}
              type="radio"
              value="Dog"
            />
            Dog
          </label>
        </div>
        <div
          className="radio"
        >
          <label>
            <input
              checked={false}
              onChange={[Function]}
              type="radio"
              value="Horse"
            />
            Horse
          </label>
        </div>
      </form>
    </div>
  </div>
  <div
    className="Text"
  >
    <div
      className="sectionContainer"
    >
      <h3>
        Text
      </h3>
      <form>
        <div
          className="radio active"
        >
          <label>
            <input
              checked={true}
              onChange={[Function]}
              type="radio"
              value="Cat"
            />
            Cat
          </label>
        </div>
        <div
          className="radio"
        >
          <label>
            <input
              checked={false}
              onChange={[Function]}
              type="radio"
              value="Dog"
            />
            Dog
          </label>
        </div>
        <div
          className="radio"
        >
          <label>
            <input
              checked={false}
              onChange={[Function]}
              type="radio"
              value="Horse"
            />
            Horse
          </label>
        </div>
      </form>
    </div>
  </div>
  <div
    className="Sound"
  >
    <div
      className="sectionContainer"
    >
      <h3>
        Sound
      </h3>
      <form>
        <div
          className="radio active"
        >
          <label>
            <input
              checked={true}
              onChange={[Function]}
              type="radio"
              value="Cat"
            />
            Cat
          </label>
        </div>
        <div
          className="radio"
        >
          <label>
            <input
              checked={false}
              onChange={[Function]}
              type="radio"
              value="Dog"
            />
            Dog
          </label>
        </div>
        <div
          className="radio"
        >
          <label>
            <input
              checked={false}
              onChange={[Function]}
              type="radio"
              value="Horse"
            />
            Horse
          </label>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`Sidebar section renders correctly 1`] = `
<div
  className="sectionContainer"
>
  <h3 />
  <form>
    <div
      className="radio active"
    >
      <label>
        <input
          checked={true}
          onChange={[Function]}
          type="radio"
          value="Cat"
        />
        Cat
      </label>
    </div>
    <div
      className="radio"
    >
      <label>
        <input
          checked={false}
          onChange={[Function]}
          type="radio"
          value="Dog"
        />
        Dog
      </label>
    </div>
    <div
      className="radio"
    >
      <label>
        <input
          checked={false}
          onChange={[Function]}
          type="radio"
          value="Horse"
        />
        Horse
      </label>
    </div>
  </form>
</div>
`;
